### PR TITLE
Qualify field accesses with "this" when necessary

### LINKED ISSUE #258: Collection property named 'elements' fails
Collection style properties named "elements" defined on a @FreeBuilder annotated class generate code that does not compile. The various `addElements` methods that get generated use the name `elements` as the argument, which causes naming collisions. I figure all that would need to be done is to ensure that references to the backing array list are made with `this` where applicable.

### PR DESCRIPTION
**Pull request by chrisalice**
Add new FieldAccess excerpt, and new static methodBody method on Block taking parameter names; together, these let us qualify field accesses with `this.` when and only when such qualification is necessary to avoid collisions. (I left the `this.` qualifiers in constructors because I feel it reads better, even though it's not technically necessary.)

`Metadata.Property` now has a derived `field` property, of type FieldAccess, that is used instead of `name` whenever the field is being referenced. The `name` property is used whenever a variable of the same name is wanted, and for string constants (e.g. Jackson annotations).

I have double-checked the changes by hacking FieldAccess to prefix _all_ field accesses with `this.` (catching one case where I'd left it in the code by mistake) and to arbitrarily rename all fields (catching a couple of cases where I'd used `name` instead of `field` by mistake).

This fixes #258.

### CODE REVIEW

File: src/test/java/org/inferred/freebuilder/processor/ListPrefixlessPropertyTest.java [hunk 1]
```diff
@@ -919,6 +919,58 @@ public void testCompilesWithoutWarnings() {
   }
 
   @Test
+  public void testCanNameBoxedPropertyElements() {
```
Comment (j-baker) [inline src/test/java/org/inferred/freebuilder/processor/ListPrefixlessPropertyTest.java]: why are these two different?

Comment (j-baker) [reply src/test/java/org/inferred/freebuilder/processor/ListPrefixlessPropertyTest.java]: oh, you're just testing all your specialized code

File: src/main/java/org/inferred/freebuilder/processor/BuildableProperty.java [hunk 2]
```diff
@@ -312,21 +308,21 @@ public void addMergeFromBuilder(Block code, String builder) {
   public void addSetBuilderFromPartial(Block code, String builder) {
     if (partialToBuilderMethod == PartialToBuilderMethod.TO_BUILDER_AND_MERGE) {
       code.add("%s.%s().mergeFrom(%s.toBuilder());",
-          builder, getBuilderMethod(property), setter(property), property.getName());
+          builder, getBuilderMethod(property), property.getField());
     } else {
       code.add("%s.%s().mergeFrom(%s);",
-          builder, getBuilderMethod(property), setter(property), property.getName());
```
Comment (j-baker) [inline src/main/java/org/inferred/freebuilder/processor/BuildableProperty.java]: ew

File: src/main/java/org/inferred/freebuilder/processor/CodeGenerator.java [hunk 3]
```diff
@@ -197,12 +201,12 @@ private static void addMergeFromBuilderMethod(SourceBuilder code, Metadata metad
         .addLine(" * Does not affect any properties not set on the input.")
         .addLine(" */")
         .addLine("public %1$s mergeFrom(%1$s template) {", metadata.getBuilder());
-    Block body = new Block(code);
+    Block body = Block.methodBody(code, "template");
```
Comment (j-baker) [inline src/main/java/org/inferred/freebuilder/processor/CodeGenerator.java]: worth pulling out these magic strings and above (template, value)

File: src/main/java/org/inferred/freebuilder/processor/CodeGenerator.java [hunk 4]
```diff
@@ -297,18 +301,20 @@ private static void addValueType(SourceBuilder code, Metadata metadata) {
...
     for (Property property : metadata.getProperties()) {
-      property.getCodeGenerator().addValueFieldDeclaration(code, property.getName());
+      property.getCodeGenerator().addValueFieldDeclaration(code, property.getField());
     }
     // Constructor
     code.addLine("")
         .addLine("  private %s(%s builder) {",
             metadata.getValueType().getSimpleName(),
             metadata.getGeneratedBuilder());
+    Block body = methodBody(code, "builder");
```
Comment (j-baker) [inline src/main/java/org/inferred/freebuilder/processor/CodeGenerator.java]: and builder

File: src/main/java/org/inferred/freebuilder/processor/CodeGenerator.java (hunk 3 above)
Comment (chrisalice) [reply src/main/java/org/inferred/freebuilder/processor/CodeGenerator.java]: As discussed offline, I think in this case this reduces readability (%s in the addLines above) without improving refactorability (only used in this method)

**Pull request merged**